content/blog/2023/05/camel-debezium-quarkus/index.md

```diff
@@ -0,0 +1,158 @@
+- [Debezium Oracle Connector Component](https://camel.apache.org/components/latest/debezium-oracle-component.html) (So far it is still not supported in Quarkus)
+
+Now that we have introduced the different technologies we are going to use, let's start building our application.
+
+You can find the complete [source code](https://github.com/mikeintoch/quarkus-debezium-camel) of the example on GitHub.
+
+`Step 1.` Create a quarkus application using maven, with the dependencies needed to build our application.
+```bash
+mvn io.quarkus:quarkus-maven-plugin:2.15.3.Final:create \                                                                                          
+    -DprojectGroupId=dev.mikeintoch \
+    -DprojectArtifactId=quarkus-debezium-camel \
+    -Dextensions="camel-quarkus-core,camel-quarkus-debezium-postgres,camel-quarkus-direct,camel-quarkus-rest,camel-quarkus-jsonpath,camel-quarkus-jackson,camel-quarkus-mongodb,camel-quarkus-jpa,quarkus-hibernate-orm,quarkus-agroal"
+```
+
+`Step 2.` Configure Debezium to connect to the PostgreSQL database and capture the changes. Create a new file called `application.properties` and add the following properties:
+```properties
+#Camel Debezium Properties
+camel.component.debezium-postgres.offset-storage=org.apache.kafka.connect.storage.FileOffsetBackingStore
+camel.component.debezium-postgres.offset-storage-file-name=/tmp/offset.dat
+camel.component.debezium-postgres.database-server-name=my_db_server
+camel.component.debezium-postgres.database-hostname=localhost
+camel.component.debezium-postgres.database-dbname=my_db
+camel.component.debezium-postgres.database-user=my_username
+camel.component.debezium-postgres.database-password=my_password
+camel.component.debezium-postgres.database-port=5432
+camel.component.debezium-postgres.plugin-name=pgoutput
+camel.component.debezium-postgres.schema-include-list=public
+camel.component.debezium-postgres.table-include-list=public.customer
+camel.component.debezium-postgres.offset-flush-interval-ms=10000
+```
+
+`Step 3.` Create a Camel route to process the changes captured by Debezium. Create a new file named `DebeziumRoute.java` and add the following code:
+```java
+import java.util.Map
+
+import org.apache.camel.builder.RouteBuilder;
+
+public class DebeziumRoute extends RouteBuilder {
+
+    @Override
+    public void configure() throws Exception {
+
+        from("debezium-postgres:my_connector")
+        .log("${body}")
+        .choice()
+          .when(header("CamelDebeziumOperation").isEqualTo("c"))
+              .process(exchange -> {
+                  // Transform the data to a format suitable for MongoDB
+                  Map<String, Object> data = exchange.getIn().getBody(Map.class);
+                  data.remove("updated_at");
+                  data.remove("created_at");
+                  data.remove("phone");
+                  exchange.getIn().setBody(data);
```

Comment:
Is-it really needed to set it again as the message body since the map is already the message body?

Reply:
Debezium component use Kafka Connect’s Struct type. I use this processor to transform to a Map Object manipulate it and after that send to MongoDB. On second thought I will change it to a ConvertBody to simplify the example.